server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -96,6 +97,11 @@ public void execute(SearchContext context) {
         FieldsVisitor fieldsVisitor = createStoredFieldsVisitor(context, storedToRequestedFields);
 
         try {
+            SearchLookup lookup = new SearchLookup(context.mapperService(), context.getQueryShardContext()::getForField);
```

Comment:
One issue here is that the runtime scripts don't have access to the shared _source. It seems like the `SearchLookup` that's passed to `fielddataBuilder` will always be the one on `QueryShardContext`? 

It actually seems confusing that `SearchLookup` lets you perform `getForField` to get field data, but then this method uses a potentially different `SearchLookup` to create it.

Reply:
Good call! I'll dig into it.